### ISSUE TITLE
Using Wayland SDL video driver causes severe banding

### ISSUE THREAD

**Issue by myownfriend** [sm64pc/sm64pc#475]
**Describe the bug**
The game seems to run at 16bpp(or some other limited bit-depth) without dithering.

**To Reproduce**
Run the game with `SDL_VIDEODRIVER=wayland`. Walk around to any area with a lot of smooth gradients. It's very noticeable when swimming in Jolly Roger Bay.

**Expected behavior**
The game should be running in 24bpp. This works when running the game through XWayland or X.org directly.

**Screenshots**
Wayland
![wayland](https://user-images.githubusercontent.com/7650024/128094491-e2cd8663-1b50-4bc3-9f1a-13d13403ae51.png)
XWayland
![xwayland](https://user-images.githubusercontent.com/7650024/128263338-727cade4-5c36-49e9-983c-1acaa43ee5d2.png)

**Desktop (please complete the following information):**
 Fedora 34

**Comment by fgsfdsfgs** [comment]
I don't know what the cause of this would be. Perhaps it needs color depth specified explicitly through `SDL_GL_SetAttribute`?

**Comment by parkerlreed** [comment]
I think I can see it even in the starting area (Banding on the green)
![Screenshot_20210804_171433](https://user-images.githubusercontent.com/841440/128255833-1dc38b89-ad65-4473-86c2-9842765570f1.png)

**Comment by parkerlreed** [comment]
Xwayland for comparison
![Screenshot_20210804_171709](https://user-images.githubusercontent.com/841440/128256128-8797b849-7e81-422f-835c-40f77d4bb4ea.png)

**Comment by myownfriend** [comment]
Yup. It's pretty much visible every where where you would expect it to be. So basically any place with smooth, low-contrast gradients which applies to a lot of the game. Places with fog or dark places make it more apparent which is why it's so bad underwater.

I was also wondering if the bit depth needs to be set explicitly but I don't know why that would be required for Wayland and not XWayland.
